src/wallet/wallet.cpp

```diff
@@ -687,8 +687,10 @@ void CWallet::MarkDirty()
     }
 }
 
-bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFromLoadWallet, CWalletDB* pwalletdb)
+bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFromLoadWallet)
 {
+    CWalletDB walletdb(strWalletFile);
```

Comment:
@jonasschnelli I will eventually get to that point, but it's going to take a significant amount of work to get there.